src/openrct2/world/Map.cpp

```diff
@@ -703,7 +703,7 @@ bool MapCoordIsConnected(const TileCoordsXYZ& loc, uint8_t faceDirection)
         }
         else
         {
-            if (loc.z == tileElement->BaseHeight)
+            if (loc.z == tileElement->BaseHeight && (tileElement->AsPath()->GetEdges() & (faceDirection << 1)))
```

Comment:
Shouldn't this be the other way around? `(1 << faceDirection)`. Also depending on what `faceDirection` actually holds, you may need to add 2 to flip the direction: `(1 << ((faceDirection + 2) & 3))`

Reply:
You're right, I had the formula backwards. It doesn't seem like inverting the direction was needed, though. I tested it with a building facing in all 4 directions and it worked. I made sure the path in front of each building was connected in every other direction so that the only unconnected part was to the building.